core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/scheduling/CostBasedRegionPlanGenerator.scala

```diff
@@ -267,12 +278,19 @@ class CostBasedRegionPlanGenerator(
           ) // Edges in chain with blocking edges should not be materialized
         }
 
-        val unvisitedNeighborStates = candidateEdges
+        var unvisitedNeighborStates = candidateEdges
           .map(edge => currentState + edge)
           .filter(neighborState =>
             !visited.contains(neighborState) && !queue.contains(neighborState)
           )
 
+        if (oEarlyStop) {
+          // Any descendant state of a schedulable state is not worth exploring.
+          unvisitedNeighborStates = unvisitedNeighborStates.filter(neighborState =>
+            !schedulableStates.exists(ancestorState => ancestorState.subsetOf(neighborState))
+          )
```

Comment:
do a set diff

Reply:
As discussed, doing a set diff is not more efficient.